gui-tk/wiring.tcl

```diff
@@ -520,21 +520,35 @@ proc doubleButton {x y} {
         editItem
     }
 }
-
+# for ticket 1062, new hierarchy of context menu access on mouse right click: wires, items and background canvas.
 bind .wiring.canvas <<contextMenu>> {
-    if [getItemAt %x %y] {
+    if [getWireAt %x %y] {
+        wireContextMenu %X %Y  	
+    } elseif [getItemAt %x %y] {
         switch [minsky.canvas.item.classType] {
             GodleyIcon {rightMouseGodley %x %y %X %Y}
             Group {rightMouseGroup %x %y %X %Y}
             default {contextMenu %x %y %X %Y}
-        }
-    } elseif [getWireAt %x %y] {
-        wireContextMenu %X %Y
+		}
     } else {
         canvasContext %x %y %X %Y
     }
 }
 
+#bind .wiring.canvas <<contextMenu>> {
+#    if [getItemAt %x %y] {
+#        switch [minsky.canvas.item.classType] {
+#            GodleyIcon {rightMouseGodley %x %y %X %Y}
+#            Group {rightMouseGroup %x %y %X %Y}
+#            default {contextMenu %x %y %X %Y}
+#        }
+#    } elseif [getWireAt %x %y] {
+#        wireContextMenu %X %Y
+#    } else {
+#        canvasContext %x %y %X %Y
+#    }
+#}
+
```

Comment:
Should remove the dead code

Reply:
Sorry, I keep forgetting that I don't need to include the previous code for comparison as Github already takes care of that. It's gone now.